examples/fluids/smartsim_regression_framework.py

```diff
@@ -23,6 +24,14 @@
 fluids_example_dir = Path(__file__).parent.absolute()
 
 
+def getOpenSocket():
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.bind(('', 0))
+    addr = s.getsockname()
+    s.close()
```

Comment:
So technically there is still a race condition here because this port could be snatched up by a different process in the time window between when it's closed and when `create_database` opens it. I don't know if one can pass an open socket to `create_database`. But I think it's very unlikely that this race condition will be triggered so we can probably ignore it (but keep in mind if we needed it to be rigorously race-free).

Reply:
Yeah, I had tried to hand `PORT=0` to see if it would automatically get the socket itself, but it just makes the database unreadable for some reason.

But yes, I think the race condition is unlikely and, importantly, inconsequential; if it happens, just rerun the CI pipeline. This isn't a problem for production runs.